Doc/library/json.rst

```diff
@@ -364,6 +364,16 @@ Basic Usage
    .. versionchanged:: 3.9
       The keyword argument *encoding* has been removed.
 
+   .. note::
+
+      As mandated by :rfc:`8259`, keys in JSON key-value pairs are always of
+      type :class:`str`. Therefore, dictionary keys obtained by
+      deserializing JSON objects will always be strings.
+      For Example,
+      ``json.loads('{"42":"spam"}')`` returns the dictionary
+      ``{'42': 'spam'}``,but ``json.loads('{42:"spam"}')``
+      will raise a exception,because it is not a correct
+      JSON text
```

Comment:
```suggestion
   .. note::

      As mandated by :rfc:`8259`, JSON keys must be :class:`str` objects.
      In particular, ``json.loads('{"42": "spam"}')`` returns ``{'42': 'spam'}``,
      but ``json.loads('{42: "spam"}')`` fails since ``42`` is not a valid JSON key.

```

Reply:
Alternatively, how about moving the note for json.dumps() to module-level? or just restate that keys in JSON must be strings?